gammapy/maps/region/geom.py

```diff
@@ -301,7 +301,25 @@ def solid_angle(self):
         if self.region is None:
             raise ValueError("Region definition required.")
 
-        area = self.region.to_pixel(self.wcs).area
+        # compound regions do not implement area()
+        # so we use the mask represenation and estimate the area
+        # from the pixels in the mask using oversampling
+        if isinstance(self.region, CompoundSkyRegion):
+            # oversample by a factor of ten
+            oversampling = 10.0
+            wcs = WcsGeom.create(
```

Comment:
Here is a nitpicky thing: `RegionGeom` also takes optionally a `wcs` object as input, which defines a custom projection, if users want this. Here you create a new one, which is not necessarily aligned (likely not...). However you can just use `self.to_wcs_geom()`, which compute the minimal "enclosing" WCS geometry. Instead of counting pixels then you could also sum up the bin wise solid angle returned by `WcsGeom.solid_angle()`.

Reply:
So do you suggest to add an oversampling option to `to_wcs_geom`? I didn't use it because I need to create a different but similar `WCSGeom` with finer bins.

Reply:
Ah, there is also `to_binsz_wcs`, which seems to work fine.

Reply:
Yes, or you can use `WcsGeom.upsample()` as well.